### PR TITLE
add iops retrieval in python-agent

### PR DESCRIPTION
**Pull request by jimengliu**
1. assuming fio tool has been run inside /var/lib/docker/tmp folder using file operation, not block operation. Run 2 times, with read I/O and write I/O each.
2. assuming the fio runs will create result files as "read.json" and "write.json" at /var/lib/rancher/state.
3. the code is to get iops for both read and write and let ping reply carry it back
@cjellick @ibuildthecloud

### CODE REVIEW

File: cattle/plugins/docker/compute.py [hunk 1]
```diff
@@ -763,6 +765,25 @@ def _setup_hostname(self, create_config, instance):
...
+    def _setup_iops(self, config, instance):
+        log.info('setting up iops for docker container named: %s',
+                 instance.name)
+        try:
+            for readIops in instance.data['fields']['blkioDeviceReadIOps']:
+                readIopsDic = {}
+                readIopsDic['Path'] = readIops['Path']
+                readIopsDic['Rate'] = int(readIops['Rate'])
+                log.info("appending readIops: %s", str(readIopsDic))
+                config['BlkioDeviceReadIOps'] = [readIopsDic]
```
Comment (cjellick) [inline cattle/plugins/docker/compute.py]: If there is more than one item in the `instance.data['fields']['blkioDeviceReadIOps']` list, aren't you stomping on it here?

File: cattle/plugins/docker/compute.py [hunk 2]
```diff
@@ -763,6 +765,25 @@ def _setup_hostname(self, create_config, instance):
...
+            for writeIops in instance.data['fields']['blkioDeviceWriteIOps']:
+                writeIopsDic = {}
+                writeIopsDic['Path'] = writeIops['Path']
+                writeIopsDic['Rate'] = int(writeIops['Rate'])
+                log.info("appending writeIops: %s", writeIopsDic)
+                config['BlkioDeviceWriteIOps'] = [writeIopsDic]
```
Comment (cjellick) [inline cattle/plugins/docker/compute.py]: same as above

File: cattle/plugins/docker/compute.py (hunk 1 above)
Comment (jimengliu) [inline cattle/plugins/docker/compute.py]: nice catch, I was testing one item. This line "config['BlkioDeviceReadIOps'] = [readIopsDic]" needs to be append. (my log already saying that, but code does not do). I will change this.

**Comment by cjellick** [comment]
@jimengliu can you add tests to test_docker for the iops fields?

You can base the test off of one of the other ones in there. The general pattern of the tests is this:
1. Set the field in the cattle event json. This simulates what cattle would send to python-agent
2. Execute the event so that python-agent creates a container with the field set
3. Verify that the docker inspect of the container has the field set properly

A relatively simple test that you can copy and modify might be: https://github.com/rancher/python-agent/blob/master/tests/test_docker.py#L502. Pick whatever one you want, I just chose this simple it seemed like a pretty simple example.

**Pull request closed without merging**